lib/cocoapods/installer/project_cache/project_cache_analyzer.rb

```diff
@@ -89,6 +89,10 @@ def analyze
 
           pod_targets_to_generate = (changed_pod_targets + added_pod_targets + dirty_pod_targets).uniq
 
+          # If one target in a pod needs generation, then all of the targets do.
+          # Otherwise we end up with the rest of the targets ungenerated.
```

Comment:
I would change this comment to something more along the lines:
```
Since multi xcodeproj generation groups targets by `pod_name`, we need to also append the sibling `PodTarget` instances under the same `pod_name` if any have changed to make sure we end up with the correct set of targets.
```
Feel free to rephrase, but the main idea is that the bug occurs because of how multi xcode proj generation works and that we need to make sure the set of pod targets to generate includes those siblings.